src/server.rs

```diff
@@ -69,6 +71,69 @@ fn replace_string_in_range(
     contents
 }
 
+fn function_defines(
+    name: &str,
+    params: &Vec<FunctionParameter>,
+) -> bool {
+    params.iter().any(|param| param.key.name == name)
+}
+
+fn is_scope(name: &str, n: Rc<walk::Node<'_>>) -> bool {
+    let mut dvisitor = DefinitionFinderVisitor::new(name.to_string());
+    walk::walk(&mut dvisitor, n.clone());
+    let state = dvisitor.state.borrow();
+
+    state.node.is_some()
+}
+
+fn find_references(
+    uri: lsp::Url,
+    result: NodeFinderResult,
+) -> Vec<lsp::Location> {
+    let mut locations: Vec<lsp::Location> = vec![];
+
+    if let Some(node) = result.node {
+        let name = match node.as_ref() {
+            walk::Node::Identifier(ident) => &ident.name[..],
+            walk::Node::IdentifierExpr(ident) => &ident.name[..],
+            _ => return locations,
+        };
+
+        let mut path_iter = result.path.iter().rev();
+        let scope: Option<Rc<walk::Node>> =
```

Comment:
I think you can potentially refactor this too in the same way.

```
let scope = match path_iter.find_map(...) {
    Some(n) => n.to_owned(),
    None => return Vec::new(),
};
```

Then you don't need the `if let` below.